bundles/org.openhab.binding.luxtronikheatpump/src/main/java/org/openhab/binding/luxtronikheatpump/internal/LuxtronikHeatpumpHandler.java

```diff
@@ -0,0 +1,344 @@
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.thing.binding.ThingHandlerCallback;
+import org.openhab.core.thing.binding.builder.ThingBuilder;
+import org.openhab.core.thing.type.ChannelTypeUID;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.openhab.core.types.State;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link LuxtronikHeatpumpHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Stefan Giehl - Initial contribution
+ */
+@NonNullByDefault
+public class LuxtronikHeatpumpHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(LuxtronikHeatpumpHandler.class);
+    private final Set<ScheduledFuture<?>> scheduledFutures = new HashSet<>();
+    private static final int RETRY_INTERVAL = 60;
+    private boolean tiggerChannelUpdate = false;
+    private final LuxtronikTranslationProvider translationProvider;
+
+    public LuxtronikHeatpumpHandler(Thing thing, LuxtronikTranslationProvider translationProvider) {
+        super(thing);
+        this.translationProvider = translationProvider;
+    }
+
+    @Override
+    public void updateState(String channelID, State state) {
+        super.updateState(channelID, state);
+    }
+
+    @Override
+    public void updateProperty(String name, String value) {
+        super.updateProperty(name, value);
+    }
+
+    public void setStatusConnectionError() {
+        LuxtronikHeatpumpConfiguration config = getConfigAs(LuxtronikHeatpumpConfiguration.class);
+
+        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                "couldn't establish network connection [host '" + config.ipAddress + "']");
+    }
+
+    public void setStatusOnline() {
+        updateStatus(ThingStatus.ONLINE);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        String channelId = channelUID.getIdWithoutGroup();
+        logger.debug("Handle command '{}' for channel {}", command, channelId);
+        if (command == RefreshType.REFRESH) {
+            // ignore resresh command as channels will be updated automatically
+            return;
+        }
+
+        HeatpumpChannel channel;
+
+        try {
+            channel = HeatpumpChannel.fromString(channelId);
+        } catch (InvalidChannelException e) {
+            logger.debug("Channel '{}' could not be found for thing {}", channelId, thing.getUID());
+            return;
+        }
+
+        if (channel.isWritable().equals(Boolean.FALSE)) {
+            logger.debug("Channel {} is a read-only channel and cannot handle command '{}'", channelId, command);
+            return;
+        }
+
+        if (command instanceof QuantityType) {
+            command = new DecimalType(((QuantityType<?>) command).floatValue());
```

Comment:
Did you consider that the return value of `floatValue()` will be Fahrenheit if the user uses Fahrenheit? You can convert it to a unit you expect by calling `((QuantityType<?>) command).toUnit()`.

Reply:
good point. Didn't think of that. will change it